antiope-s3/src/Antiope/S3.hs

```diff
@@ -54,6 +56,14 @@ toS3Uri :: BucketName -> ObjectKey -> Text
 toS3Uri (BucketName b) (ObjectKey k) =
   "s3://" <> b <> "/" <> k
 
+fromS3Uri :: Text -> Maybe S3Location
+fromS3Uri uri = do
+  puri <- parseURI (unpack uri)
+  auth <- puri & uriAuthority
+  let b = pack $ auth & uriRegName       -- URI lib is pretty weird
+  let k = pack $ drop 1 $ puri & uriPath
+  pure $ S3Location (BucketName b) (ObjectKey k)
```

Comment:
Ideally I would like a `FromText` as well, but I don't necessarily need it right now